backend/testfiles/execution/stdlib/result.dark

```diff
@@ -101,6 +101,117 @@ PACKAGE.Darklang.Stdlib.Result.map2_v0
+  (PACKAGE.Darklang.Stdlib.Result.Result.Error "error3")
+  (PACKAGE.Darklang.Stdlib.Result.Result.Ok 1)
+  (fun (a, b, c, d) -> a - b - c - d) = PACKAGE.Darklang.Stdlib.Result.Result.Error "error1"
+
+PACKAGE.Darklang.Stdlib.Result.map4_v0
+  (PACKAGE.Darklang.Stdlib.Result.Result.Ok 10)
+  (PACKAGE.Darklang.Stdlib.Result.Result.Ok 1)
+  (PACKAGE.Darklang.Stdlib.Result.Result.Ok 1)
+  (PACKAGE.Darklang.Stdlib.Result.Result.Error "error4")
+  (fun (a, b, c, d) -> a - b - c - d) = PACKAGE.Darklang.Stdlib.Result.Result.Error "error4"
+
+PACKAGE.Darklang.Stdlib.Result.map4_v0
+  (PACKAGE.Darklang.Stdlib.Result.Result.Ok 10)
+  (PACKAGE.Darklang.Stdlib.Result.Result.Ok 1)
+  (PACKAGE.Darklang.Stdlib.Result.Result.Ok 1)
+  (PACKAGE.Darklang.Stdlib.Result.Result.Ok 2)
+  (fun (a, b, c, d) -> a - b - c - d) = PACKAGE.Darklang.Stdlib.Result.Result.Ok 6
+
+
+PACKAGE.Darklang.Stdlib.Result.map5_v0
+  (PACKAGE.Darklang.Stdlib.Result.Result.Error "error1")
+  (PACKAGE.Darklang.Stdlib.Result.Result.Error "error2")
+  (PACKAGE.Darklang.Stdlib.Result.Result.Error "error3")
+  (PACKAGE.Darklang.Stdlib.Result.Result.Error "error4")
+  (PACKAGE.Darklang.Stdlib.Result.Result.Error "error5")
+  (fun (a, b, c, d, e) -> a - b - c - d - e) = PACKAGE.Darklang.Stdlib.Result.Result.Error "error1"
+
+PACKAGE.Darklang.Stdlib.Result.map5_v0
+  (PACKAGE.Darklang.Stdlib.Result.Result.Error "error1")
+  (PACKAGE.Darklang.Stdlib.Result.Result.Error "error2")
+  (PACKAGE.Darklang.Stdlib.Result.Result.Error "error3")
+  (PACKAGE.Darklang.Stdlib.Result.Result.Error "error4")
+  (PACKAGE.Darklang.Stdlib.Result.Result.Ok 1)
+  (fun (a, b, c, d, e) -> a - b - c - d - e) = PACKAGE.Darklang.Stdlib.Result.Result.Error "error1"
+
+PACKAGE.Darklang.Stdlib.Result.map5_v0
+  (PACKAGE.Darklang.Stdlib.Result.Result.Ok 10)
+  (PACKAGE.Darklang.Stdlib.Result.Result.Ok 1)
+  (PACKAGE.Darklang.Stdlib.Result.Result.Ok 1)
+  (PACKAGE.Darklang.Stdlib.Result.Result.Ok 2)
+  (PACKAGE.Darklang.Stdlib.Result.Result.Error "error5")
+  (fun (a, b, c, d, e) -> a - b - c - d - e) = PACKAGE.Darklang.Stdlib.Result.Result.Error "error5"
+
+PACKAGE.Darklang.Stdlib.Result.map5_v0
+  (PACKAGE.Darklang.Stdlib.Result.Result.Ok 10)
+  (PACKAGE.Darklang.Stdlib.Result.Result.Ok 1)
+  (PACKAGE.Darklang.Stdlib.Result.Result.Ok 1)
+  (PACKAGE.Darklang.Stdlib.Result.Result.Ok 2)
+  (PACKAGE.Darklang.Stdlib.Result.Result.Ok 3)
+  (fun (a, b, c, d, e) -> a - b - c - d - e) = PACKAGE.Darklang.Stdlib.Result.Result.Ok 3
+
+
+
+PACKAGE.Darklang.Stdlib.Result.mapWithDefault_v0
+  (PACKAGE.Darklang.Stdlib.Result.Result.Error "test")
+  5
+  (fun x -> x + 1) = 5
+
+PACKAGE.Darklang.Stdlib.Result.mapWithDefault_v0
+  (PACKAGE.Darklang.Stdlib.Result.Result.Ok 6)
+  5
+  (fun x -> x + 1) = 7
+
+PACKAGE.Darklang.Stdlib.Result.mapWithDefault_v0
+  (PACKAGE.Darklang.Stdlib.Result.Result.Error "test1")
+  (PACKAGE.Darklang.Stdlib.Result.Result.Error "test2")
+  (fun x -> x + 1) = PACKAGE.Darklang.Stdlib.Result.Result.Error "test2"
```

Comment:
I would expect `test1` to be the result here - why did you go with `test2`?

Reply:
mapWithDefault applies the function to a result if it's `Result.Ok.` If the result is a `Result.Error`, it returns the default value. In this case, the default value is also an Error, which is why test2 was returned. 
Were you expecting a different behavior?

Reply:
Gotcha. This seems correct, thanks for the clarification.